packages/js/product-editor/src/contexts/validation-context/README.md

```diff
@@ -0,0 +1,39 @@
+# Validations and error handling
+
+This directory contains some components and hooks used for validations in the product editor.
+
+See [Product Editor Development Handbook](../../../../../../docs/product-editor-development/README.md)
+for more information.
+
+## Add a link to error snackbar
+
+### What happens when there is an error in a field?
+
+1. Every field [will be validated](https://github.com/woocommerce/woocommerce/blob/trunk/packages/js/product-editor/src/contexts/validation-context/validation-provider.tsx#L87-L110).
+
+2. An object with the [fields with errors](https://github.com/woocommerce/woocommerce/blob/trunk/packages/js/product-editor/src/contexts/validation-context/validation-provider.tsx#L74) will be returned.
+
+3. If, for instance, the name field is empty, the validation will fail and will throw an object like this:
+
+    ```javascript
+    { message: 'Product name is required.'; context: [block id]; validatorId: [prefix + block id] }
+    ```
+
+4. This is the result of the [name validator](https://github.com/woocommerce/woocommerce/blob/trunk/packages/js/product-editor/src/blocks/product-fields/name/edit.tsx#L84-L101) and the [validatorId addition](https://github.com/woocommerce/woocommerce/blob/trunk/packages/js/product-editor/src/contexts/validation-context/validation-provider.tsx#L69).
+
+5. Then, [the props needed](https://github.com/woocommerce/woocommerce/blob/trunk/packages/js/product-editor/src/components/header/publish-button/publish-button.tsx#L65-L71) to show the error snackbar will be returned.
```

Comment:
Hmm I think we should also re-phrase this. Maybe something like this:

```
3. If an error is present on the form we will show an error snackbar with the error message. We will actually include a **show error** link if the field with the relevant error is not visible ( like on another tab ). Clicking the **show error** link will direct users to the relevant field.
     - We create this link by making use of the `context` property ( the block id ), this makes use of the `useBlocksHelper()` hook to get the parent tab id. We can do this by making use of the `core/block-editor` store and using `getBlockParentsByBlockName` ( link to relevant code )
```

Reply:
Done in [be38da9](https://github.com/woocommerce/woocommerce/pull/49723/commits/be38da928a6429fd51eed853c2fe3eebe15e30f9)